pkg/chain/ethereum/stake_monitor.go

```diff
@@ -1,67 +1,21 @@
 package ethereum
 
 import (
-	"fmt"
-	"github.com/keep-network/keep-core/pkg/operator"
-	"math/big"
-
-	"github.com/ethereum/go-ethereum/common"
-	relaychain "github.com/keep-network/keep-core/pkg/beacon/relay/chain"
 	"github.com/keep-network/keep-core/pkg/chain"
+	"github.com/keep-network/keep-core/pkg/operator"
 )
 
-type ethereumStakeMonitor struct {
-	ethereum *ethereumChain
+type stakeMonitor struct {
+	chain *Chain
 }
 
-func (esm *ethereumStakeMonitor) HasMinimumStake(
+// TODO: Real implementation with v2 contracts.
+func (sm *stakeMonitor) HasMinimumStake(
```

Comment:
@tomaszslabon Your #3060 will need this function. The proposed approach is: the `firewall` package should define an interface it is expecting, or import `chain.StakeMonitor`. I think the latter is fine for the time being. Per #3010, we no longer want to check `HasMinimumStake` but `HasStakeDelegation` (we should rename this function). And the logic checking staking provider <> operator mapping and the TokenStaking roles should be checked inside `HasStakeDelegated`. 

I think we should check both `RandomBeacon` and `WalletRegistry` for staking provider <> operator mapping. We want to have one network of nodes for both applications.